src/jobsautoreport/models.py

```diff
@@ -24,8 +24,40 @@ class ReportInterval(Enum):
     MONTH = "month"
 
 
-class MachineMetrics(BaseModel):
+class Metric(BaseModel):
+    def is_zero(self) -> bool:
+        for cost in self.dict().values():
+            if cost > 0:
+                return False
+        return True
+
+
+class MachineMetrics(Metric):
     metrics: dict[str, float]
 
     def is_zero(self) -> bool:
         return sum(self.metrics.values()) == 0
+
+
+class JobTypeMetrics(Metric):
+    presubmit: float = 0.0
+    periodic: float = 0.0
+    postsubmit: float = 0.0
+    batch: float = 0.0
+
+
+class JobMetrics(Metric):
+    successes: int
+    failures: int
+
+    @property
+    def total(self) -> int:
+        return self.successes + self.failures
+
+    @property
+    def failure_rate(self) -> float:
+        return 0 if self.total == 0 else (self.failures / self.total) * 100
```

Comment:
mathematically it should not be defined (like 'N/A'), because 0 is a valid failure rate only if you have any kind of attempts 
```suggestion
        return None if self.total == 0 else (self.failures / self.total) * 100
```